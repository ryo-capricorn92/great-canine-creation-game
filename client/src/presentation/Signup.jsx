import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Label = styled.label`
  font-size: 15px;
  font-weight: bold;
  margin-right: 5px;
`;

const LabelCell = styled.td`
  padding: 3px;
  text-align: right;
`;

const InputCell = styled.td`
  padding: 3px;
`;

const SubmitCell = styled.td.attrs({
  colSpan: '2',
})`
  padding: 5px;
`;

class Signup extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      usernameConfirm: false,
    };
  }

  render() {
    return (
      <Container>
        <form>
          <table>
            <tbody>
              <tr>
                <LabelCell>
                  <Label htmlFor="username">Username:</Label>
                </LabelCell>
                <InputCell>
                  <input name="username" />
                </InputCell>
              </tr>
              <tr>
                <LabelCell>
                  <Label htmlFor="password">Password:</Label>
                </LabelCell>
                <InputCell>
                  <input name="password" type="password" />
                </InputCell>
              </tr>
              <tr>
                <LabelCell>
                  <Label htmlFor="email">Email:</Label>
                </LabelCell>
                <InputCell>
                  <input name="email" type="email" />
                </InputCell>
              </tr>
              <tr>
                <LabelCell>
                  <Label htmlFor="email_confirm">Confirm Email:</Label>
                </LabelCell>
                <InputCell>
                  <input name="email_confirm" type="email" />
                </InputCell>
              </tr>
              <tr>
                <LabelCell>
                  <Label htmlFor="birthday">Birthday:</Label>
                </LabelCell>
                <InputCell>
                  <input name="birthday" type="date" />
                </InputCell>
              </tr>
              <tr>
                <SubmitCell>
                  <input type="submit" value="Submit" />
                </SubmitCell>
              </tr>
            </tbody>
          </table>
        </form>
      </Container>
    );
  }
}

export default Signup;
