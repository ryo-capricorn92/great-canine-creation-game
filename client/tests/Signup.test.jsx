import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Signup from '../src/presentation/Signup';

test('Signup renders', () => {
  const wrapper = mount(<Signup />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
