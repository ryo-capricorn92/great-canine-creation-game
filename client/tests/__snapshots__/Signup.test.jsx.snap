// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders 1`] = `
<Signup>
  <styled.div>
    <div
      className="sc-bdVaJa cxMafI"
    >
      <form>
        <table>
          <tbody>
            <tr>
              <styled.td>
                <td
                  className="sc-htpNat gZOKTT"
                >
                  <styled.label
                    htmlFor="username"
                  >
                    <label
                      className="sc-bwzfXH fGBoem"
                      htmlFor="username"
                    >
                      Username:
                    </label>
                  </styled.label>
                </td>
              </styled.td>
              <styled.td>
                <td
                  className="sc-bxivhb kFSGKt"
                >
                  <input
                    name="username"
                  />
                </td>
              </styled.td>
            </tr>
            <tr>
              <styled.td>
                <td
                  className="sc-htpNat gZOKTT"
                >
                  <styled.label
                    htmlFor="password"
                  >
                    <label
                      className="sc-bwzfXH fGBoem"
                      htmlFor="password"
                    >
                      Password:
                    </label>
                  </styled.label>
                </td>
              </styled.td>
              <styled.td>
                <td
                  className="sc-bxivhb kFSGKt"
                >
                  <input
                    name="password"
                    type="password"
                  />
                </td>
              </styled.td>
            </tr>
            <tr>
              <styled.td>
                <td
                  className="sc-htpNat gZOKTT"
                >
                  <styled.label
                    htmlFor="email"
                  >
                    <label
                      className="sc-bwzfXH fGBoem"
                      htmlFor="email"
                    >
                      Email:
                    </label>
                  </styled.label>
                </td>
              </styled.td>
              <styled.td>
                <td
                  className="sc-bxivhb kFSGKt"
                >
                  <input
                    name="email"
                    type="email"
                  />
                </td>
              </styled.td>
            </tr>
            <tr>
              <styled.td>
                <td
                  className="sc-htpNat gZOKTT"
                >
                  <styled.label
                    htmlFor="email_confirm"
                  >
                    <label
                      className="sc-bwzfXH fGBoem"
                      htmlFor="email_confirm"
                    >
                      Confirm Email:
                    </label>
                  </styled.label>
                </td>
              </styled.td>
              <styled.td>
                <td
                  className="sc-bxivhb kFSGKt"
                >
                  <input
                    name="email_confirm"
                    type="email"
                  />
                </td>
              </styled.td>
            </tr>
            <tr>
              <styled.td>
                <td
                  className="sc-htpNat gZOKTT"
                >
                  <styled.label
                    htmlFor="birthday"
                  >
                    <label
                      className="sc-bwzfXH fGBoem"
                      htmlFor="birthday"
                    >
                      Birthday:
                    </label>
                  </styled.label>
                </td>
              </styled.td>
              <styled.td>
                <td
                  className="sc-bxivhb kFSGKt"
                >
                  <input
                    name="birthday"
                    type="date"
                  />
                </td>
              </styled.td>
            </tr>
            <tr>
              <styled.td>
                <td
                  className="sc-ifAKCX gYKwRG"
                  colSpan="2"
                >
                  <input
                    type="submit"
                    value="Submit"
                  />
                </td>
              </styled.td>
            </tr>
          </tbody>
        </table>
      </form>
    </div>
  </styled.div>
</Signup>
`;
